src/main/java/aquality/selenium/core/elements/DefaultElementStateProvider.java

```diff
@@ -86,4 +110,21 @@ public boolean waitForEnabled(Duration timeout) {
     public boolean waitForNotEnabled(Duration timeout) {
         return isElementInDesiredCondition(elementNotEnabled(), timeout);
     }
+
+    private boolean doAndLogWaitForState(BooleanSupplier waitingAction, String conditionKeyPart, Duration timeout)
+    {
+        if (Duration.ZERO.equals(timeout))
+        {
+            return waitingAction.getAsBoolean();
+        }
+
+        logElementState("loc.wait.for.state", conditionKeyPart);
+        boolean result = waitingAction.getAsBoolean();
```

Comment:
can we move this var at the first line in the method and use it for 'ZERO' branch too

Reply:
no we can not, the order of logging-action-logging does matter here